src/components/PieChart.js

```diff
@@ -0,0 +1,182 @@
+const React = require('react');
+const ReactDOM = require('react-dom');
+const d3 = require('d3');
+
+const StyleConstants = require('../constants/Style');
+
+const PieChart = React.createClass({
+  propTypes: {
+    animateDuration: React.PropTypes.number,
+    animationTypeOnLoad: React.PropTypes.oneOf(['roll', 'pop']),
+    arcWidth: React.PropTypes.number,
+    colors: React.PropTypes.array,
+    data: React.PropTypes.array.isRequired,
+    height: React.PropTypes.number,
+    hoverExpandDistance: React.PropTypes.number,
+    onLabelClick: React.PropTypes.func,
+    onMouseEnter: React.PropTypes.func,
+    onMouseLeave: React.PropTypes.func,
+    onPieClick: React.PropTypes.func,
+    showDataLabel: React.PropTypes.bool,
+    style: React.PropTypes.object,
+    width: React.PropTypes.number
+  },
+
+  getDefaultProps () {
+    return {
+      animationTypeOnLoad: 'roll',
+      animateDuration: 500,
+      arcWidth: 2,
+      colors: [StyleConstants.Colors.PRIMARY].concat(d3.scale.category20().range()),
+      height: 350,
+      hoverExpandDistance: 10,
+      onLabelClick () {},
+      onMouseEnter () {},
+      onMouseLeave () {},
+      onPieClick () {},
+      showDataLabel: true,
+      width: 300
+    };
+  },
+
+  componentDidMount () {
+    const dom = ReactDOM.findDOMNode(this.refs.chart);
+
+    this._renderChart(dom);
+  },
+
+  _renderShowLabels () {
+    const styles = this.styles();
+
+    return (
+      <div
+        className='mx-donutchart-data'
+        onClick={this.props.onLabelClick}
+        style={styles.center}
+      >
+        {this.props.children}
+      </div>
+    );
+  },
+
+  _renderChart (dom) {
+    const { animateDuration, arcWidth, height, width, hoverExpandDistance, data, colors } = this.props;
+    const thisGlobal = this;
```

Comment:
Yeah, that is a good question. I tried doing that but the whole issue around passing global `this` and local `this` to different functions make this component very hard to read and confusing. It is more of an issue with D3 and how it handles animations. I thought it would be best to leave it as, so it is easer to read it from top to bottom.
@cerinman @jmophoto Any thoughts?